dotCMS/src/main/java/com/dotcms/publisher/pusher/AuthCredentialPushPublishUtil.java

```diff
@@ -0,0 +1,201 @@
+                new PushPublishAuthenticationToken(publishingEndPointOptional.get()) :
+                PushPublishAuthenticationToken.INVALID_TOKEN;
+    }
+
+    private PushPublishAuthenticationToken getFromJWTToken(HttpServletRequest request) {
+        try {
+            final Optional<JWToken> jwTokenOptional =
+                    JsonWebTokenAuthCredentialProcessorImpl.getInstance().processJWTAuthHeader(request);
+
+            if (!jwTokenOptional.isPresent()){
+                return PushPublishAuthenticationToken.INVALID_TOKEN;
+            }
+
+            final Optional<User> optionalUser = jwTokenOptional.get().getActiveUser();
+            if (!optionalUser.isPresent()){
+                return PushPublishAuthenticationToken.INVALID_TOKEN;
+            }
+
+            return new PushPublishAuthenticationToken(jwTokenOptional.get());
+        } catch(IncorrectClaimException e){
+            final String claimName = e.getClaimName();
+
+            return Claims.EXPIRATION.equals(claimName) ? PushPublishAuthenticationToken.EXPIRE_TOKEN :
+                    PushPublishAuthenticationToken.INVALID_TOKEN;
+        }
+    }
+
+    private Optional<PublishingEndPoint> getPublishingEndPointDotCMSToken(final HttpServletRequest request)
+            throws DotDataException, IOException {
+        final String remoteIP = request.getRemoteHost();
+        final PublishingEndPoint publishingEndPoint =
+                APILocator.getPublisherEndPointAPI().findEnabledSendingEndPointByAddress(remoteIP);
+
+        Optional<String> key = PushPublisher.retriveEndpointKeyDigest(publishingEndPoint);
+        if(!key.isPresent()) {
+            return Optional.empty();
+        }
+
+        final String token = getTokenFromRequest(request);
+
+        return token.equals( key.get() ) ? Optional.of(publishingEndPoint) : Optional.empty();
+    }
+
+    private String getTokenFromRequest(final HttpServletRequest request) {
+        final String authorizationHeader = request.getHeader(ContainerRequest.AUTHORIZATION);
+
+        if (StringUtils.isNotEmpty(authorizationHeader) && authorizationHeader.trim()
+                .startsWith(BEARER)) {
+
+            return authorizationHeader.substring(BEARER.length());
+        } else {
+            throw new IllegalArgumentException("Bearer Authorization header expected");
+        }
+    }
+
+    private Optional<String> getJWTToken(final PublishingEndPoint endpoint) throws IOException {
+        return PushPublisher.retriveEndpointKey(endpoint);
+    }
+
+    public static class PushPublishAuthenticationToken {
+
+        public static final PushPublishAuthenticationToken EXPIRE_TOKEN = new PushPublishAuthenticationToken(true, false);
+        public static final PushPublishAuthenticationToken INVALID_TOKEN = new PushPublishAuthenticationToken(false, true);
+        private final boolean tokenExpired;
+        private final boolean tokenInvalid;
+
+        private boolean usingJWTTokenWay = false;
```

Comment:
Codacy found an issue: [Private field 'usingJWTTokenWay' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)